sentry-android-core/src/main/java/io/sentry/android/core/gestures/SentryGestureListener.java

```diff
@@ -0,0 +1,238 @@
+
+@ApiStatus.Internal
+public final class SentryGestureListener implements GestureDetector.OnGestureListener {
+
+  private final @NotNull WeakReference<Window> windowRef;
+  private final @NotNull IHub hub;
+  private final @NotNull SentryAndroidOptions options;
+
+  private final ScrollState scrollState = new ScrollState();
+
+  public SentryGestureListener(
+    final @NotNull WeakReference<Window> windowRef,
+    final @NotNull IHub hub,
+    final @NotNull SentryAndroidOptions options
+  ) {
+    this.windowRef = windowRef;
+    this.hub = hub;
+    this.options = options;
+  }
+
+  public void onUp(final @NotNull MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onUp");
+    final View scrollTarget = scrollState.targetRef.get();
+    if (decorView == null || scrollTarget == null) {
+      return;
+    }
+
+    if (scrollState.type == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.INFO, "Unable to define scroll type. No breadcrumb captured.");
+      return;
+    }
+
+    final String direction = scrollState.calculateDirection(motionEvent);
+    addBreadcrumb(scrollTarget, scrollState.type,
+      Collections.singletonMap("direction", direction));
+    scrollState.reset();
+  }
+
+  @Override
+  public boolean onDown(final @Nullable MotionEvent motionEvent) {
+    if (motionEvent == null) {
+      return false;
+    }
+    scrollState.reset();
+    scrollState.startX = motionEvent.getX();
+    scrollState.startY = motionEvent.getY();
+    return false;
+  }
+
+  @Override
+  public boolean onSingleTapUp(final @Nullable MotionEvent motionEvent) {
+    final View decorView = ensureWindowDecorView("onSingleTapUp");
+    if (decorView == null || motionEvent == null) {
+      return false;
+    }
+
+    @SuppressWarnings("Convert2MethodRef")
+    final @Nullable View target =
+      ViewUtils.findTarget(decorView, motionEvent.getX(), motionEvent.getY(),
+        view -> ViewUtils.isViewTappable(view));
+
+    if (target == null) {
+      options
+        .getLogger()
+        .log(SentryLevel.INFO, "Unable to find click target. No breadcrumb captured.");
+      return false;
+    }
+
+    addBreadcrumb(target, "click");
+    return false;
+  }
+
+  @Override
+  public boolean onScroll(
+    final @Nullable MotionEvent firstEvent,
+    final @Nullable MotionEvent currentEvent,
+    final float distX,
+    final float distY
+  ) {
+    final View decorView = ensureWindowDecorView("onScroll");
+    if (decorView == null || firstEvent == null) {
+      return false;
+    }
+
+    if (scrollState.type == null) {
+      @SuppressWarnings("Convert2MethodRef")
+      final @Nullable View target =
+        ViewUtils.findTarget(decorView, firstEvent.getX(), firstEvent.getY(),
+          view -> ViewUtils.isViewScrollable(view));
+
+      if (target == null) {
+        options
+          .getLogger()
+          .log(SentryLevel.INFO, "Unable to find scroll target. No breadcrumb captured.");
```

Comment:
Similar to above, perhaps debug. Or if this seems like a bug, possibly WARN instead?